# HG changeset patch
# User Rik <rik@octave.org>
# Date 1466552339 25200
#      Tue Jun 21 16:38:59 2016 -0700
# Node ID 17457503ba952dd50cd1806b2f0ebe7a3525e287
# Parent  112b20240c8791a40a9739ad4b3dd492b13c9043
gendoc.pl: Minor changes for readability and speed.

* gendoc.pl: Don't bother to capture reference in regexp which is never used.
use \Q,\E pairs to turn off specialness of regexp characters temporarily to
make patterns easier to read.

diff --git a/libinterp/gendoc.pl b/libinterp/gendoc.pl
--- a/libinterp/gendoc.pl
+++ b/libinterp/gendoc.pl
@@ -15,18 +15,17 @@
 # for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 unless (@ARGV > 1) { die "Usage: $0 SRCDIR src-file1 ..." }
 
-$srcdir = $ARGV[0];
-shift;
+$srcdir = shift (@ARGV);
 
 print <<__END_OF_MSG__;
 ### DO NOT EDIT!
 ###
 ### This file is generated automatically from Octave source files.
 ### Edit source files directly and run make to update this file.
 
 __END_OF_MSG__
@@ -44,38 +43,38 @@ FILE: foreach $fname (@ARGV)
 
   open (SRC_FH, $src_fname) or die "Unable to open $src_fname";
 
   @func_list = ();
   @docstr = ();
 
   LINE: while (<SRC_FH>)
   {
-    if (/^\s*DEF(CONSTFUN|UN|UN_DLD|UNX|UNX_DLD)\s*\(/)
+    if (/^\s*DEF(?:CONSTFUN|UN|UN_DLD|UNX|UNX_DLD)\s*\(/)
     {
       ($func) = /\("?(\w+)"?,/;
       unless ($func) { die "Unable to parse $src_fname at line $.\n" }
       push (@func_list, $func);
 
-      if (<SRC_FH> =~ /\s*doc:\s+\/\*\s+-\*- texinfo -\*-\s*$/)
+      if (<SRC_FH> =~ m#\s*doc:\s+\Q/*\E\s+\Q-*- texinfo -*-\E\s*$#)
       {
         $str = "-*- texinfo -*-\n";
         $reading_docstring = 1;
       }
       else
       {
         print STDERR "gendoc.pl: undocumented function $func from $fname\n";
         push (@docstr, "Undocumented.");
       }
     }
     elsif ($reading_docstring)
     {
       if (/^.*\s+\*\/\s*\)\s*$/)
       {
-        s/\s+\*\/\s*\)\s*$//;
+        s#\s+\*/\s*\)\s*$##;
         push (@docstr, $str . $_);
         $reading_docstring = 0;
       }
       else
       {
         $str .= $_;
       }
     }
